jstest/emulate/twisted_server.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2018-present Samsung Electronics Co., Ltd. and other contributors
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from threading import Thread
+
+from twisted.cred.portal import Portal
+from twisted.conch.ssh.factory import SSHFactory
+from twisted.internet import reactor
+from twisted.conch.ssh.keys import Key
+from twisted.cred.checkers import FilePasswordDB
+from twisted.conch.interfaces import IConchUser
+from twisted.conch.avatar import ConchUser
+from twisted.conch.ssh.channel import SSHChannel
+from twisted.internet.protocol import Protocol
+from twisted.conch.ssh.session import SSHSessionProcessProtocol, wrapProtocol
+
+
+class SimpleSession(SSHChannel):
+    name = 'session'
+
+    # pylint: disable=invalid-name
+    def dataReceived(self, data):
+        cmd = str(data)
+
+        if 'iotjs_build_info' in cmd:
+            self.write('{ "builtins": {}, "features": {}, "stability": "stable" }\r\n')
+        else:
+            exitcode = 1 if 'run_fail' in cmd else 0
+            self.write('{"output": "",'
+                       ' "memstat": {"heap-system": "n/a", "heap-jerry": "n/a", "stack": "n/a"},'
+                       ' "exitcode": %d}\r\n' % exitcode)
```

Comment:
`exitcode` can be replaced with `int('fail' in cmd)`. Please note that JerryScript uses `fail` folder name for the expected failure tests.

Reply:
A better solution to format JSON string is to create an object and use `json.dumps`:

```python
if 'iotjs_build_info' in cmd:
    result = {
        'builtins': {},
        'features': {}
        'stability': 'stable'
    }

else:
    result = {
        'output': 'Hello from emulated device.',
        'memstat': {
            'heap-system': 'n/a',
            'heap-jerry': 'n/a',
            'stack': 'n/a'
        },
        'exitcode': int('fail' in cmd)
    }

self.write(json.dumps(result) + '\r\n')
```